packages/@aws-cdk/aws-ec2-alpha/lib/route.ts

```diff
@@ -663,4 +787,46 @@ function routerTypeToPropName(routerType: RouterType) {
     [RouterType.VPC_PEERING_CONNECTION]: 'vpcPeeringConnectionId',
     [RouterType.VPC_ENDPOINT]: 'vpcEndpointId',
   })[routerType];
+}
+
+/**
+ * Validates if the provided IPv4 CIDR block overlaps with existing subnet CIDR blocks within the given VPC.
+ *
+ * @param requestorVpc The VPC of the requestor.
+ * @param acceptorVpc The VPC of the acceptor.
+ * @returns True if the IPv4 CIDR block overlaps with existing subnet CIDR blocks, false otherwise.
+ * @internal
+ */
+function validateVpcCidrOverlap(requestorVpc: IVpcV2, acceptorVpc: IVpcV2): boolean {
```

Comment:
You can probably add this function to the [util.ts](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-ec2-alpha/lib/util.ts) file

Reply:
It should be private

Reply:
done! [2b4c5c6](https://github.com/aws/aws-cdk/pull/31645/commits/2b4c5c6a80d0bb26306b6d94826113a1580e6846)